engine/src/main/java/org/hibernate/validator/internal/engine/messageinterpolation/ParameterTermResolver.java

```diff
@@ -38,6 +39,16 @@ public String interpolate(Context context, String expression) {
 		return resolvedExpression;
 	}
 
+	private Object getVariable(Context context, String parameter) {
+		if ( context instanceof HibernateMessageInterpolatorContext ) {
```

Comment:
I wondered if it could be worth it to create the aggregation earlier. I decided against it because `getMessageParameters()` is preexisting and I'm a bit worried about changing the definition of it.